### PR TITLE
Authentication implemented

### PR DESCRIPTION
**Pull request by kgajriaAgarwal**

### CODE REVIEW

File: src/Pages/Authentication/Login/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+          } else{
+            setAlertContent({_id: uuid(), isShow:true, type:'ERROR', content:"Unexpected error.Login Failed"})
+          }
+        })
+        .catch((response) => {
+          if (response.data && response.data.error) {
+            setAlertContent({_id: uuid(), isShow:true, type:'ERROR', content:"Unexpected error.Login Failed"})
+          }
+        })
+    }
```
Comment (amantulsyan35) [inline src/Pages/Authentication/Login/Login.jsx:48]: can use try catch, trying to understand why have you used .then .catch here

Comment (kgajriaAgarwal) [reply src/Pages/Authentication/Login/Login.jsx:48]: ok! ty for the suggestion

File: src/Pages/Authentication/Login/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                        <img src={blitheLogo} className="blithe-logo" alt="blithe-logo"/>
+                        <form>
+                            <p className="heading-md heading">Login</p>
+                            <InputField 
+                                type='email'
+                                title='Email'
+                                name='email'
+                                placeholder='Enter email'
+                                value={loginData?.email? loginData.email : ''}
+                                onChange={(e)=>setLoginData(prevData => {return  {...prevData, email:e.target.value}})}
```
Comment (mehulsatardekar) [inline src/Pages/Authentication/Login/Login.jsx:69]: here before return, you can remove the curly brackets so u don't have to write explicitly return for it

File: src/Pages/Authentication/Login/Login.jsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                            />
+                             <InputField 
+                                type={showPassword?'text':'password'}
+                                title='Password'
+                                name='password'
+                                placeholder='Enter password'
+                                showPassword={showPassword}
+                                handlePassword = {handlePassword}
+                                value={loginData?.password? loginData.password : ''}
+                                onChange={(e)=>setLoginData(prevData => {return  {...prevData, password:e.target.value}})}
```
Comment (mehulsatardekar) [inline src/Pages/Authentication/Login/Login.jsx:79]: do the same changes for this also

File: src/Pages/Authentication/SignUp/SignUp.jsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                            />
+                            <InputField 
+                                type={showPassword?'text':'password'}
+                                title='Password'
+                                name='password'
+                                placeholder='Enter password'
+                                showPassword={showPassword}
+                                handlePassword = {handlePassword}
+                                value={sigUpData?.password? sigUpData.password : ''}
+                                onChange={(e)=>setSignUpData(prevData => {return  {...prevData, password:e.target.value}})}
```
Comment (mehulsatardekar) [inline src/Pages/Authentication/SignUp/SignUp.jsx:103]: The return part wont need it here  i believe, you can do the same like 
onChange{(e) => setSignUpData( prevData=> {...prevData, password:e.target.value}  )  }

**Review by mehulsatardekar — APPROVED** [review]
The Rest thing looks good and folders are nicely structured

File: src/Pages/Authentication/SignUp/SignUp.jsx (hunk 4 above)
Comment (kgajriaAgarwal) [reply src/Pages/Authentication/SignUp/SignUp.jsx:103]: its showing error when not using return here !

File: src/Pages/Authentication/Login/Login.jsx (hunk 3 above)
Comment (kgajriaAgarwal) [reply src/Pages/Authentication/Login/Login.jsx:79]: its showing error when not using return here !

File: src/Pages/Authentication/Login/Login.jsx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+import '../index.css';
+import { Input, InputField } from '../../../Shared';
+import blitheCollage from '../../../Assets/Images/Blithe-collage.jpg';
+import blitheLogo from '../../../Assets/Images/Blithe-logo.png';
+import axios from 'axios';
+import { useNavigate } from "react-router-dom";
+import { actionLogin } from '../../../Helpers/Services/actions';
+import { v4 as uuid } from "uuid";
+import { useAlert } from '../../../Context/AlertContext';
+
```
Comment (Ankur9669) [inline src/Pages/Authentication/Login/Login.jsx:12]: Suggestion: you can import all files in one single file

File: src/Pages/Authentication/Login/Login.jsx [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    const [showPassword , setShowPassword ] = useState(false);
+    const [loginData, setLoginData] = useState(
+        {   
+            email:'',
+            password:''  
+        })
+
+    //Handle show/hide password
+    const handlePassword = () =>{
+        setShowPassword(!showPassword)
```
Comment (Ankur9669) [inline src/Pages/Authentication/Login/Login.jsx:26]: setShowPassword(showPassword => !showPassword)

File: src/Pages/Authentication/Login/Login.jsx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+
+    const handleLogin = async (e) =>{
```
Comment (Ankur9669) [inline src/Pages/Authentication/Login/Login.jsx:30]: You can remove async from here

File: src/Pages/Authentication/Login/Login.jsx [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+        e.preventDefault();
+
+        actionLogin(loginData)
```
Comment (Ankur9669) [inline src/Pages/Authentication/Login/Login.jsx:33]: Suggestion: If you are async then use await here

File: src/Pages/Authentication/Login/Login.jsx [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+                                onChange={(e)=>setLoginData(prevData => {return  {...prevData, password:e.target.value}})}
+                            />
+
+                            <div className="remember">
+                                <label>
+                                    <Input id='rememberUser' type="checkbox" name ='remember' checked={true}/>
+                                    Remember me!
+                                </label>
+                                <br/>
+                                <Link  to="/forgot-password" className="btn-link-forgot-password" target="_blank">Forgot Password ??</Link>
```
Comment (Ankur9669) [inline src/Pages/Authentication/Login/Login.jsx:88]: fix: I think you should remove target=_blank

File: src/Pages/Authentication/SignUp/SignUp.jsx [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+            firstName:'',
+            lastName:'',
+            password:'',
+            confirmPassword:'',  
+        })
+    //here err is for the validation purpose, that is yet to be implemented
+
+    //Handle show/hide password
+    const handlePassword = () =>{
+        setShowPassword(!showPassword)
```
Comment (Ankur9669) [inline src/Pages/Authentication/SignUp/SignUp.jsx:30]: Same suggestion as above

File: src/Pages/Authentication/SignUp/SignUp.jsx [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    //Handle show/hide confirm password
+    const handleConfirmPassword = () =>{
+        setShowConfirmPassword(!showConfirmPassword)
```
Comment (Ankur9669) [inline src/Pages/Authentication/SignUp/SignUp.jsx:35]: Here also same suggestion

File: src/Pages/Authentication/index.css [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+    color: var(--secondary-color);
+    text-decoration: underline;
+    font-weight: 600;
+}
+
+.btn-link-forgot-password:hover{
+    color: var(--primary-bg);
+}
+
+@media (max-width:768px){
```
Comment (Ankur9669) [inline src/Pages/Authentication/index.css:127]: fix: use @media only screen and (max-width: 768px) here

File: src/Shared/Input/InputField.jsx [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+import passwordHideIcon from '../../Assets/Icons/password-hide.png';
+import { Input } from './Input';
+
+export const InputField = (props) => {
+    // <!-- Input Field Error -->
+    
+    const handleIcon = () => {
+        if(props.name === 'password'){
+            return <img className="eye-icon" 
+                    src={props.showPassword?passwordShowIcon:passwordHideIcon}
```
Comment (Ankur9669) [inline src/Shared/Input/InputField.jsx:13]: Suggestion: You can destructure the props and use them

File: src/Shared/Sidebar/Sidebar.jsx [hunk 14]
```diff
@@ -18,14 +20,14 @@ export const Sidebar = () => {
...
                             <span className="links_name">Home</span>
-                        </a>
-                        <a href="#">
+                        </Link>
+                        <Link to="/videos/:categoryId">
                             <MdExplore color='white' size='1.5rem' className='nav-icon' />
                             <span className="links_name">Explore </span>
-                        </a>
+                        </Link>
                         <a href="#">
```
Comment (Ankur9669) [inline src/Shared/Sidebar/Sidebar.jsx:31]: Can replace with link tag

**Review by Ankur9669 — COMMENTED** [review]
Most the things looks good to me, i have suggested some changes you can work on them

**Pull request merged**